src/main/java/kitchenpos/menu/domain/MenuProductEntity.java

```diff
@@ -0,0 +1,79 @@
+package kitchenpos.menu.domain;
+
+import kitchenpos.product.domain.ProductEntity;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.Objects;
+
+@Table(name = "menu_product")
+@Entity
+public class MenuProductEntity {
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.IDENTITY)
+  private Long seq;
+
+  @ManyToOne
+  @JoinColumn(name = "menu_id", nullable = false)
+  private MenuEntity menu;
+
+  @ManyToOne
+  @JoinColumn(name = "product_id", nullable = false)
+  private ProductEntity product;
+
+  @Embedded
+  private Quantity quantity;
+
+  protected MenuProductEntity() {
+  }
+
+  public MenuProductEntity(ProductEntity product, long quantity) {
+    this.product = product;
+    this.quantity = Quantity.from(quantity);
+  }
+
+  public MenuProductEntity(Long seq, ProductEntity product, long quantity) {
+    this.seq = seq;
+    this.product = product;
+    this.quantity = Quantity.from(quantity);
+  }
+
+  public Long getSeq() {
+    return seq;
+  }
+
+  public MenuEntity getMenu() {
+    return menu;
+  }
+
+  public ProductEntity getProduct() {
+    return product;
+  }
+
+  public long getQuantity() {
+    return quantity.getValue();
+  }
+
+  public void defineParentMenu(MenuEntity menuEntity) {
+    this.menu = menuEntity;
+  }
+
+  public Price calculateAmount() {
+    BigDecimal productPrice = product.getPrice();
+    return Price.from(productPrice.multiply(quantity.getBigDecimalValue()));
+  }
```

Comment:
getter 를 통해 price 의 값을 직접 가져오는 것 보다는 객체에 메세지를 보내보면 어떨까요? 😄 
```suggestion
  public Price calculateAmount() {
    return product.calculatePrice(quantity);
  }
```

Reply:
productEntity는 다른 product의 aggregate의 객체이고, quantity는 menu agregate의 객체인데요,
(quantity는 그래서 다른 aggregate에서 참조할 수 없게 default 접근제한자를 사용해두었습니다.)
product에게 quantity를 넘기게 되면 다른 aggregate 끼리 직접참조하게되는 케이스가 되지 않을까요?? 🤔

Reply:
Quantity 는 수량을 나타내는 VO 로 보여지네요!
또한 Quantity 는 현재는 menu 에서만 사용하지만 수량 이라는 의미로 보았을 때 추후에 다른 aggregate 에서도 사용될 수도 있을 것 같은데요 따로 참조하지 못하게 해두신 이유가 있으실까요? 😄 

product 에게 quantity 넘기는 건이 다른 aggregate 끼리 직접참조하게 되는 케이스라고 보여지지는 않는데
태엽님의 생각은 어떠신가요? 😄